tests/_test_process_executor.py

```diff
@@ -1034,3 +1019,65 @@ def test_child_env_executor(self):
         assert var_child == var_value
 
         executor.shutdown(wait=True)
+
+    def test_viztracer_profiler(self):
+        # Check that viztracer profiler is initialzed in workers when
+        # installed.
+        viztracer = pytest.importorskip("viztracer")
+
+        def check_viztracer_active():
+            tracer = viztracer.get_tracer()
+            if tracer is None:
+                return False
+            return tracer.enable
+
+        active_in_main_process = check_viztracer_active()
+        with self.executor_type(1, context=self.context) as e:
+            active_in_child_process = e.submit(check_viztracer_active).result()
+        assert active_in_main_process == active_in_child_process
+
+        if not active_in_main_process:
+            tracer = viztracer.VizTracer()
+            try:
+                tracer.start()
+                with self.executor_type(1, context=self.context) as e:
+                    assert e.submit(check_viztracer_active).result()
+            finally:
+                tracer.stop()
+
+            # Once the tracer has been stopped, should be no side effect on
+            # workers started in new executors.
+            with self.executor_type(1, context=self.context) as e:
+                assert not e.submit(check_viztracer_active).result()
```

Comment:
I have added this additional check as an explicit non-regression check for the failure that was caused by a side effect.

Reply:
:+1: